proposals/NNNN-unlock-existential-types-for-all-protocols.md

```diff
@@ -0,0 +1,405 @@
+# Unlock Existentials with Fixed Associated Types
+
+* Proposal: [SE-NNNN](NNNN-unlock-existential-types-for-all-protocols.md)
+* Authors: [Anthony Latsis](https://github.com/AnthonyLatsis), [Filip Sakel](https://github.com/filip-sakel), [Suyash Srijan](https://github.com/theblixguy)
+* Review Manager: TBD
+* Status: **Awaiting Review**
+* Implementation: [apple/swift#33767](https://github.com/apple/swift/pull/33767)
+
+## Introduction
+
+Swift currently offers the ability for protocols that meet certain criteria to be used as types. Trying to use an unsupported protocol as a type yields the error: [the protocol] can only be used as a generic constraint because it has 'Self' or associated type requirements. This proposal aims to relax this artificial constraint imposed on such protocols.
```

Comment:
> yields the following error:

I think that adding "following" is redundant; that's the function of the colon. If you think, though, that "following" makes this significantly clearer, then go ahead and change it, I don't have any strong opinions about it either.

> protocols with requirements meeting certain criteria

That's much clearer, thank you!

> Perhaps we should be a tad more specific and mention that – instead of the protocol itself – the error will now affect the ability to invoke protocol members in accordance with the criteria.

You're right, that _would_ be more accurate. 

Do you think the following looks better?

> Swift currently offers the ability for protocols with requirement meeting certain criteria to be used as types. Trying to use an unsupported protocol as a type yields the error: [the protocol] can only be used as a generic constraint because it has 'Self' or associated type requirements. This proposal aims to relax this artificial constraint, and instead constraint the availability of protocol members.